File: backend/entity/workshop/modrating.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type ModRating struct {
	gorm.Model

	Rating  string
	Review string
	PurchaseDate  time.Time
	

	GameUserID *uint
	GameUser   GameUser `gorm:"foreignKey:GameUserID"`

	ModID *uint
	Mod   Mod `gorm:"foreignKey:ModID"`
	
}

File: frontend/src/interfaces/ModRating.ts
import type { GameUser } from "./GameUser";
import type { Mod } from "./Mod";

export interface ModRating {
  ID: number;
  Rating: string;
  Review?: string;
  PurchaseDate?: string;
  GameUserID?: number;
  ModID?: number;
  GameUser?: GameUser;
  Mod?: Mod;
}